downloads/swan-lake-release-notes/swan-lake-alpha3/RELEASE_NOTE.md

```diff
@@ -2,41 +2,928 @@
+
+```ballerina
+public string name = "Ballerina";
+
+public [int, float] [a, b] = [1, 2.5];
+```
+
+##### Improvement to Annotation Attachment with Empty Mapping Constructor Expression
+
+If the type of the annotation is a mapping type for which an empty mapping constructor is valid, the mapping constructor expression is no longer mandatory in the annotation attachment.
+
+The absence of the mapping constructor expression in such an annotation attachment is equivalent to specifying a mapping constructor expression with no fields.
+```ballerina
+type Annot record {|
+    int[] i = [];
+|};
+
+public annotation Annot v1 on function;
+
+@v1 // Same as `@v1 {}`
+public function main() {
+}
+```
+
+##### Introduction of the `function` Function Type Descriptor to Represent Any Function
+
+A new `function` type descriptor has been introduced to represent all function values.
+
+```ballerina
+import ballerina/io;
+
+function add(int v1, int v2) returns int => v1 + v2;
+
+function compare(int v1, int v2) returns boolean => v1 < v2;
+
+public function main() {
+    // A variable of type `function` can hold any function value.
+    function f = add;
+    io:println("Process (add, 1, 2): ", process(f, 1, 2)); // Prints `Process (add, 1, 2): 3`
+    io:println("Process (compare, 1, 2): ", process(compare, 1, 2)); // Prints `Process (compare, 1, 2): 0`
+}
+
+function process(function func, int v1, int v2) returns int {
+    // A function of type `function` cannot be called directly.
+    // A function value assigned to a `function`-typed variable
+    // can only be called after the type is narrowed to the relevant type.
+    if (func is function (int, int) returns int) {
+        return func(v1, v2);
+    }
+    return 0;
+}
+```
+
+##### Improved lang library functions
+
+###### New `xml:text()` function
+
+This function can be used to select all the items in a sequence that are of type `xml:Text`.
+
+```ballerina
+xml name = xml `<name>Dan<middleName>Gerhard</middleName><!-- This is a comment -->Brown</name>`;
+xml:Text nameText = (name/*).text();
+io:println(nameText); // "DanBrown"
+```
+
+##### Bug Fixes
+
+To view bug fixes, see the [GitHub milestone for Swan Lake Alpha3](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+Alpha3%22+label%3AType%2FBug+label%3ATeam%2FCompilerFE).
 
 #### Runtime
--
+
+##### New runtime Java API to create errors
+
+A new runtime Java API is introduced to create errors.
+```java
+BError createError(Module module, String errorTypeName, BString message, BError cause, Object details)
+```
+The `createDistinctError` API has been deprecated and should not be used to create distinct errors. The new `createError` API can be used instead.
+
+##### Configurable Variables
+
+Improved error messages are introduced for configurable variables by including more information and the relevant `Config.toml` line numbers.
```

Comment:
We don't usually include improvements to error messages specifically like this right? @MaryamZi thoughts?

Reply:
Yeah, I guess this isn't something we usually add to the release note. @warunalakshitha wdyt?

Reply:
+1 to remove that section. @HindujaB Can u check too?

Reply:
yes. +1 to remove this

Reply:
Removed